File: leet-code/defangIPaddr.go
package main

import (
	"strings"
)

func defangIPaddr(address string) string {
	newAddress := ""
	for i := range address {
		if address[i] == '.' {
			newAddress += "[.]"
		} else {
			newAddress += string(address[i])
		}
	}
	return newAddress
}

func defangIPaddr2(address string) string {
	return strings.Replace(address, ".", "[.]", -1)
}

File: leet-code/defangIPaddr.js
/**
 * @param {string} address
 * @return {string}
 */
var defangIPaddr = function (address) {
  let newAddress = ''
  for (let i = 0; i < address.length; i++) {
    if (address[i] === '.') {
      newAddress += '[.]'
    } else {
      newAddress += address[i]
    }
  }
  return newAddress
}

var defangIPaddr = function (address) {
  const reg = new RegExp('\\.','g')
  return address.replace(reg, '[.]',)
}
console.log(defangIPaddr('1.1.1.1'))